python/python-fastapi-server/main.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+from fastapi import Depends, FastAPI, Request
+import uvicorn
+
+from fastapi.middleware.cors import CORSMiddleware
+from routers.healthcheck import healthCheckRouter
+from routers.wmswcs import wmsWcsRouter, testadaguc
+from routers.autowms import autoWmsRouter
+from routers.opendap import opendapRouter
+from routers.ogcapi import ogcApiApp
+
+from routers.middleware import FixSchemeMiddleware
+import time
+
+import logging
+from configureLogging import configureLogging
+
+configureLogging(logging)
+logger = logging.getLogger(__name__)
+
+# from adaguc import routeAdagucServer
+
+app = FastAPI()
+
+
+@app.middleware("http")
+async def add_process_time_header(request: Request, call_next):
+    start_time = time.time()
+    response = await call_next(request)
+    process_time = time.time() - start_time
+    response.headers["X-Process-Time"] = str(process_time)
+    return response
+
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
```

Comment:
Is this what we had in Flask? 
I don't think origins=* together with allows_credentials is a good idea.
Actually, according to the documentation this is not possible:
`allow_credentials - Indicate that cookies should be supported for cross-origin requests. Defaults to False. Also, allow_origins cannot be set to ['*'] for credentials to be allowed, origins must be specified.`

Do you get an error from this?

Reply:
No error seen.